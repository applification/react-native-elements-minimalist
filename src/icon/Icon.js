import React, { Component } from 'react';
import PropTypes from 'prop-types';

import getIconType from '../helpers/getIconType';

class IconElement extends Component {
  static propTypes = {
    type: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    color: PropTypes.string.isRequired,
  };

  static defaultProps = {
    type: 'feather',
    name: 'user',
    color: '#000000',
  };

  render() {
    const Icon = getIconType(this.props.type);

    return (
      <Icon
        name={this.props.name}
        testID={this.props.name}
        size={30}
        color={this.props.color}
        style={{ padding: 10 }}
      />
    );
  }
}

export default IconElement;
