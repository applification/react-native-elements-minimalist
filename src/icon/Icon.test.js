import React from 'react';
import renderer from 'react-test-renderer';

import Icon from './Icon';

describe('ICON', () => {
  it('Renders with DefaultProps', () => {
    const component = renderer.create(<Icon type="feather" />).toJSON();
    const { testID } = component.props;
    const { color } = component.props.style[0];
    const { padding } = component.props.style[1];
    const { fontFamily } = component.props.style[2];

    expect(testID).toBe('user');
    expect(color).toBe('#000000');
    expect(fontFamily).toBe('Feather');
    expect(padding).toBe(10);

    expect(component).toMatchSnapshot();
  });

  it('Renders with a user selected icon font', () => {
    const component = renderer
      .create(<Icon type="simple-line-icons" />)
      .toJSON();
    const { testID } = component.props;
    const { color } = component.props.style[0];
    const { padding } = component.props.style[1];
    const { fontFamily } = component.props.style[2];

    expect(testID).toBe('user');
    expect(color).toBe('#000000');
    expect(fontFamily).toBe('simple-line-icons');
    expect(padding).toBe(10);

    expect(component).toMatchSnapshot();
  });

  it('Renderes with a custom color', () => {
    const component = renderer.create(<Icon color="#D7235F" />).toJSON();
    const { testID } = component.props;
    const { color } = component.props.style[0];
    const { padding } = component.props.style[1];
    const { fontFamily } = component.props.style[2];

    expect(testID).toBe('user');
    expect(color).toBe('#D7235F');
    expect(fontFamily).toBe('Feather');
    expect(padding).toBe(10);

    expect(component).toMatchSnapshot();
  });
});
