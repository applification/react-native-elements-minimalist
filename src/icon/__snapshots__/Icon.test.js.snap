// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ICON Renderes with a custom color 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "#D7235F",
        "fontSize": 30,
      },
      Object {
        "padding": 10,
      },
      Object {
        "fontFamily": "Feather",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
  testID="user"
>
  
</Text>
`;

exports[`ICON Renders with DefaultProps 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "#000000",
        "fontSize": 30,
      },
      Object {
        "padding": 10,
      },
      Object {
        "fontFamily": "Feather",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
  testID="user"
>
  
</Text>
`;

exports[`ICON Renders with a user selected icon font 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "#000000",
        "fontSize": 30,
      },
      Object {
        "padding": 10,
      },
      Object {
        "fontFamily": "simple-line-icons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
  testID="user"
>
  
</Text>
`;
